www/js/metrics/metricsHelper.ts

```diff
@@ -0,0 +1,212 @@
+import { DateTime } from "luxon";
+import { formatForDisplay } from "../config/useImperialConfig";
+import { DayOfMetricData } from "./metricsTypes";
+import moment from 'moment';
+
+export function getUniqueLabelsForDays(metricDataDays: DayOfMetricData[]) {
+  const uniqueLabels: string[] = [];
+  metricDataDays.forEach(e => {
+    Object.keys(e).forEach(k => {
+      if (k.startsWith('label_')) {
+        const label = k.substring(6); // remove 'label_' prefix leaving just the mode label
+        if (!uniqueLabels.includes(label)) uniqueLabels.push(label);
+      }
+    });
+  });
+  return uniqueLabels;
+}
+
+export const getLabelsForDay = (metricDataDay: DayOfMetricData) => (
+  Object.keys(metricDataDay).reduce((acc, k) => {
+    if (k.startsWith('label_')) {
+      acc.push(k.substring(6)); // remove 'label_' prefix leaving just the mode label
+    }
+    return acc;
+  }, [] as string[])
+);
+
+export const secondsToMinutes = (seconds: number) =>
+  formatForDisplay(seconds / 60);
+
+export const secondsToHours = (seconds: number) =>
+  formatForDisplay(seconds / 3600);
+
+// segments metricsDays into weeks, with the most recent week first
+export function segmentDaysByWeeks (days: DayOfMetricData[], nWeeks?: number) {
+  const weeks: DayOfMetricData[][] = [];
+  for (let i = days?.length - 1; i >= 0; i -= 7) {
+    weeks.push(days.slice(Math.max(i - 6, 0), i + 1));
+  }
+  if (nWeeks) return weeks.slice(0, nWeeks);
+  return weeks;
+};
+
+export function formatDate(day: DayOfMetricData) {
+  const dt = DateTime.fromISO(day.fmt_time, { zone: 'utc' });
+  return dt.toLocaleString({...DateTime.DATE_SHORT, year: undefined});
+}
+
+export function formatDateRangeOfDays(days: DayOfMetricData[]) {
+  if (!days?.length) return '';
+  const firstDayDt = DateTime.fromISO(days[0].fmt_time, { zone: 'utc' });
+  const lastDayDt = DateTime.fromISO(days[days.length - 1].fmt_time, { zone: 'utc' });
+  const firstDay = firstDayDt.toLocaleString({...DateTime.DATE_SHORT, year: undefined});
+  const lastDay = lastDayDt.toLocaleString({...DateTime.DATE_SHORT, year: undefined});
+  return `${firstDay} - ${lastDay}`;
+}
+
+/* formatting data form carbon footprint calculations */
+
+//modes considered on foot for carbon calculation, expandable as needed
+const ON_FOOT_MODES = ['WALKING', 'RUNNING', 'ON_FOOT'] as const;
+
```

Comment:
as long as we use outputs beyond CLEAN_AND_RESAMPLE in the pipeline, we should not need this mapping. We should clarify that we only support cleaned/inferred/confirmed values in the dashboard so that we can simplify this logic.

Reply:
because after cleaning, all of these would turn into 'WALKING', right?

Reply:
Yup. The `RUNNING` and `ON_FOOT` are really for data that comes in directly from the phone.